app/ui-react/syndesis/src/modules/integrations/components/editor/EditorSteps.tsx

```diff
@@ -0,0 +1,97 @@
+import { getMetadataValue } from '@syndesis/api';
+import * as H from '@syndesis/history';
+import {
+  ConnectionCard,
+  ConnectionsGrid,
+  ConnectionsGridCell,
+  ConnectionSkeleton,
+} from '@syndesis/ui';
+import { WithLoader } from '@syndesis/utils';
+import * as React from 'react';
+import { Translation } from 'react-i18next';
+import { ApiError, EntityIcon } from '../../../../shared';
+import { IUIStep } from './interfaces';
+
+export interface IEditorStepsProps {
+  error: boolean;
+  loading: boolean;
+  steps: IUIStep[];
+
+  getEditorStepHref(editorStep: IUIStep): H.LocationDescriptor;
+}
+
+export class EditorSteps extends React.Component<IEditorStepsProps> {
+  public render() {
+    return (
+      <Translation ns={['connections', 'shared']}>
+        {t => (
+          <ConnectionsGrid>
+            <WithLoader
+              error={this.props.error}
+              loading={this.props.loading}
+              loaderChildren={
+                <>
+                  {new Array(5).fill(0).map((_, index) => (
+                    <ConnectionsGridCell key={index}>
+                      <ConnectionSkeleton />
+                    </ConnectionsGridCell>
+                  ))}
+                </>
+              }
+              errorChildren={<ApiError />}
+            >
+              {() => {
+                return this.props.steps.map((s, index) => {
+                  const configurationRequired =
+                    s.board &&
+                    (s.board!.notices ||
+                      s.board!.warnings ||
+                      s.board!.errors)! > 0;
+
+                  let isTechPreview = false;
+
+                  if (s.connection && s.connection!.connector) {
```

Comment:
Similarly with this block.  I'd say it'd be more worthwhile in this case to accept an `IUIStep`, `Connection` and `Connector` object.  Ideally it'd be cool if this could be reduced to:

```
const isTechPreview = figureOutTechPreviewFlag(s);
```

And then of course sort the other spots in the codebase where we're figuring this out.

Reply:
This would be good, but I think should be addressed in a separate PR.

Reply:
Why?  You're in the code now, it's probably the best time to tackle this while it's fresh, ya know?  It'd be a matter of writing functions that just return a boolean, then this block here and [this block](https://github.com/syndesisio/syndesis-react/blob/master/app/ui-react/syndesis/src/modules/connections/components/Connections.tsx#L79-L91) will do the same thing then.  Seems like a win to me 'cause you'll be reducing code.